### PR TITLE
Better select

### PR DESCRIPTION
**Pull request by sophypal**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

* **Added** the option to specify static options for the `select` input
* **Added** the option to enable local filtering for the `select` input

### CODE REVIEW

File: addon/components/inputs/select.js [hunk 1]
```diff
@@ -48,6 +48,41 @@ export default AbstractInput.extend({
...
+        value: ''
+      })
+      data = [none].concat(data)
+    }
+
+    return data
+  },
+
+  @readOnly
+  @computed('cellConfig.renderer.options.localFiltering')
```
Comment (sandersky) [inline addon/components/inputs/select.js]: I've been trying to avoid using computed decorators on nested properties to make sure Ember doesn't convert our POJOs to Ember Objects. Instead I'd just do:

```js
@readOnly
@computed('cellConfig')
isFilteringLocally (cellConfig) {
  return _.get(cellConfig, 'renderer.options.localFiltering') || false
},
```

File: addon/templates/components/frost-bunsen-input-select.hbs [hunk 2]
```diff
@@ -5,17 +5,31 @@
...
+      data=options
+      disabled=disabled
+      hook=hook
+      onChange=(action 'handleChange')
+      onFocusOut=(action 'showErrorMessage')
+      placeholder=cellConfig.placeholder
+      selectedValue=value
+    }}
+  {{else}}
+    {{frost-select
```
Comment (sandersky) [inline addon/templates/components/frost-bunsen-input-select.hbs]: It looks like the same code is in both the if and else conditions here. Can't you just completely revert the changes to this file?

File: tests/dummy/app/pods/view/renderers/documentation/select.md [hunk 3]
```diff
@@ -22,3 +22,93 @@ This is the default renderer when the model contains the property `enum` or the
   "placeholder": "Select something…"
 }
 ```
+
+### Querying Data
```
Comment (sandersky) [inline tests/dummy/app/pods/view/renderers/documentation/select.md]: You mean we are finally going to document how all of this magic works? :+1:

File: addon/components/inputs/select.js [hunk 4]
```diff
@@ -48,6 +48,41 @@ export default AbstractInput.extend({
     return false
   },
 
+  @readOnly
+  @computed('bunsenModel', 'cellConfig.renderer.options')
```
Comment (sandersky) [inline addon/components/inputs/select.js]: I'd use `@computed('bunsenModel', 'cellConfig')` instead.

File: addon/templates/components/frost-bunsen-input-select.hbs (hunk 2 above)
Comment (sophypal) [inline addon/templates/components/frost-bunsen-input-select.hbs]: One has `onInput` set and the other doesn't.

Comment (sandersky) [inline addon/templates/components/frost-bunsen-input-select.hbs]: I reread that 3 times and missed it every time. Thanks for clarifying.

Comment (sandersky) [inline addon/templates/components/frost-bunsen-input-select.hbs]: Ember really needs the spread operator for properties so we don't have to do this if/else duplication.

Comment (sandersky) [inline addon/templates/components/frost-bunsen-input-select.hbs]: Actually could you do the following?

```js
    {{frost-select
      class=valueClassName
      data=options
      disabled=disabled
      hook=hook
      onChange=(action 'handleChange')
      onFocusOut=(action 'showErrorMessage')
      onInput=(if isFilteringLocally undefined (action 'filterOptions'))
      placeholder=cellConfig.placeholder
      selectedValue=value
    }}
```

**Comment by sandersky** [comment]
:+1:

**Comment by sandersky** [comment]
:+1:
